documentation/general/dash-sonic-hld.md

```diff
@@ -1527,3 +1575,61 @@ For the example configuration above, the following is a brief explanation of loo
 		h. Third Action is Appliance Encap for id 22
 		i. Packet shall be encapsulated with Outer DIP as 100.8.1.2 and SIP as VIP of this originating appliance card with VNI of 101. 
 		j. Inbound flow shall be similar to PL and outer encap shall be of the SLB MUX and not of the NSG appliance.
+
+# 4 SONiC DASH virtual switch
+## 4.1 Motivation
+
+1. Provide a Proof of Concept (POC) for development and collaboration. Utilizing the typical SONiC workflow, we can leverage this virtual switch image to construct a virtual testbed, eliminating the need for a complete hardware device. This virtual DPU image enables the creation of a mixed hardware-software testbed or a software-only testbed, applicable to both the control plane and the data plane.
+2. Enable Continuous Integration(CI) via Azure Pipelines (Azp) for SONiC repositories, like sonic-buildimage, sonic-swss and so on.
+
+## 4.2 Architecture
+
+![BMv2 virtual SONiC](images/bmv2-virtual-sonic.svg)
+
+## 4.3 Modules
+
+### 4.3.1 BMv2 (dataplane engine)
+
+This component is the original P4 BMv2 container image, which serves as the implementation of the dataplane.
+It attaches three types of interfaces: Ethernet, eth, and cpu.
+- Ethernet is used as the system port. Protocol services like BGP and LLDP perform send/receive operations on these interfaces.
+- eth is used as the line port. These are native interfaces in KVM for communication between the inside and outside. The eth port and Ethernet port is one-to-one mapping.
+- cpu is used for the dpdk port. The dataplane APP directly manipulates the traffic on these ports.
+
+### 4.3.2 Dataplane APP
+
+In the physical DPU, this component is provided by vendors' DPDK application. However, in the virtual switch, our implementation is based on the VPP framework with the CPU interface to simulate it. This dataplane APP loads the generated shared library, saidash.so, which communicates with BMv2 via GRPC. Additionally, this component connects with sairedis through a shim SAI proxy.
+
+### 4.3.3 SAIRedis
+
+In the original virtual SONiC, SAIRedis will load the saivs.so. However, in the SONiC DASH virtual switch, it will load the sai proxy mentioned above.
+
+### 4.3.4 SWSS
+
+The SWSS on the virtual switch is almost the same as the one used in the physical switch (DPU). We don't need to make any special changes to it. However, due to some tedious historic issues, the SWSS on the virtual switch will make some extra SAI calls (like DTEL). To adapt to these calls, we will generate a corresponding mock SAI implementation in saidash.so.
+
+### 4.3.5 Underlay Service
```

Comment:
Maybe we can rename this section to "Other SONiC services". And these services will be kept, so the KVM runs the same way as how SONiC runs on the real DPU.

Reply:
Done